compiler/src/printCommon.ml

```diff
@@ -0,0 +1,92 @@
+open Format
+open Utils
+open Prog
+open Wsize
+
+module E = Expr
+
+(* -------------------------------------------------------------------- *)
+
+let pp_string0 fmt str =
+  fprintf fmt "%a" (pp_list "" pp_print_char) str
+
+(* -------------------------------------------------------------------- *)
+
+let pp_wsize fmt sz =
+  Format.fprintf fmt "%a" pp_string0 (string_of_wsize sz)
+
+(* -------------------------------------------------------------------- *)
+
+let string_of_signess s =
+  if s = Unsigned then "u" else "s"
+
+let string_of_velem s ws ve =
+  let nws = int_of_ws ws in
+  let nve = int_of_velem ve in
+  let s   = string_of_signess s in
+  asprintf "%d%s%d" (nws/nve) s nve
+
+(* -------------------------------------------------------------------- *)
+
+let string_of_cmp_ty = function
+  | E.Cmp_w (Unsigned, _) -> "u"
+  | _        -> ""
+
+let string_of_cmp_kind = function
+  | E.Cmp_w (sg, sz) -> asprintf " %d%s" (int_of_ws sz) (string_of_signess sg)
+  | E.Cmp_int -> ""
+
+let string_of_op_kind = function
+  | E.Op_w ws -> asprintf "%du" (int_of_ws ws)
+  | E.Op_int -> ""
+
+(* -------------------------------------------------------------------- *)
+
+let string_of_op1 = function
+  | E.Oint_of_word sz -> asprintf "(int /* of u%d */)" (int_of_ws sz)
+  | E.Osignext (szo, _) -> asprintf "(%ds)" (int_of_ws szo)
+  | E.Oword_of_int szo
+    | E.Ozeroext (szo, _) -> asprintf "(%du)" (int_of_ws szo)
+  | E.Olnot _ -> "!"
+  | E.Onot    -> "!"
+  | E.Oneg _ -> "-"
+
+let string_of_op2 = function
+  | E.Obeq   -> "="
+  | E.Oand   -> "&&"
+  | E.Oor    -> "||"
+  | E.Oadd _ -> "+"
+  | E.Omul _ -> "*"
+  | E.Osub _ -> "-"
+  | E.Odiv k -> "/"  ^ string_of_cmp_kind k
+  | E.Omod k -> "%"  ^ string_of_cmp_kind k
+
+  | E.Oland _ -> "&"
+  | E.Olor _ -> "|"
+  | E.Olxor _ -> "^"
+  | E.Olsr _ -> ">>"
+  | E.Olsl _ -> "<<"
+  | E.Oasr _ -> ">>s"
+
+  | E.Oeq  k -> "==" ^ string_of_op_kind k
+  | E.Oneq k -> "!=" ^ string_of_op_kind k
+  | E.Olt  k -> "<"  ^ string_of_cmp_ty k
+  | E.Ole  k -> "<=" ^ string_of_cmp_ty k
+  | E.Ogt  k -> ">"  ^ string_of_cmp_ty k
+  | E.Oge  k -> ">=" ^ string_of_cmp_ty k
+
+  | Ovadd (ve,ws) -> asprintf "+%s"  (string_of_velem Unsigned ws ve)
+  | Ovsub (ve,ws) -> asprintf "-%s"  (string_of_velem Unsigned ws ve)
+  | Ovmul (ve,ws) -> asprintf "*%s"  (string_of_velem Unsigned ws ve)
+  | Ovlsr (ve,ws) -> asprintf ">>%s" (string_of_velem Unsigned ws ve)
+  | Ovasr (ve,ws) -> asprintf ">>%s" (string_of_velem Unsigned ws ve)
+  | Ovlsl (ve,ws) -> asprintf "<<%s" (string_of_velem Signed   ws ve)
+
+(* -------------------------------------------------------------------- *)
+
+let pp_var tbl fmt x =
+  let y = Conv.var_of_cvar tbl x in
+  fprintf fmt "%s" y.v_name
+
+let pp_var_i tbl fmt x =
+  pp_var tbl fmt x.E.v_var
```

Comment:
This is used only in printLinear.ml, is it worth to put it here?

Reply:
This is in anticipation of a future change. Should I postpone this?

Reply:
No, that's ok.